### ISSUE TITLE
inlein connection refused error - request for help.

### ISSUE THREAD

**Issue by RicSwirrl** [hyPiRion/inlein#22]
This is a bit of a newbie question.... 

On OSX Sierra 10.12.4,  and Java `build 1.8.0_121-b13` when i try to run some `inlein` commands.

e.g. 
`inlein --start-daemon` or
`inlein path/to/foo.clj` 

...I get an error like the below. 

```
Exception in thread "main" java.net.ConnectException: Connection refused (Connection refused)
	at java.net.PlainSocketImpl.socketConnect(Native Method)
	at java.net.AbstractPlainSocketImpl.doConnect(AbstractPlainSocketImpl.java:350)
	at java.net.AbstractPlainSocketImpl.connectToAddress(AbstractPlainSocketImpl.java:206)
	at java.net.AbstractPlainSocketImpl.connect(AbstractPlainSocketImpl.java:188)
	at java.net.SocksSocketImpl.connect(SocksSocketImpl.java:392)
	at java.net.Socket.connect(Socket.java:589)
	at java.net.Socket.connect(Socket.java:538)
	at java.net.Socket.<init>(Socket.java:434)
	at java.net.Socket.<init>(Socket.java:211)
	at inlein.client.ServerConnection.tryConnect(ServerConnection.java:77)
	at inlein.client.ServerConnection.ensureConnected(ServerConnection.java:23)
	at inlein.client.tasks.StartDaemon.run(StartDaemon.java:20)
	at inlein.client.Main.main(Main.java:32)
```

inlein itself  has permissions `-rwxr-xr-x@` and is on my `PATH`

Using `sudo` makes it work, but I don't understand why. 🤷‍♂️ 

It used to work ok for me a couple of weeks ago, but I hadn't run the task for a while.  So, some pointers on what the error means, and what might have changed for this to occur would be appreciated! Since it last worked for me, I think I've upgraded new patch versions of Java and OS X.

Thanks!

**Comment by cflaming** [comment]
Ran into this as well after a restart on mac. I got around it by compiling `inlein` from source. Had some issues getting `test/install.sh` working without changing GNU `cp  -t` and `readlink -f` to something compatible with BSD/Mac.

**Comment by markmathis** [comment]
This looks like #15. A work around is to delete the ~/.inlein/port file.

**Comment by cflaming** [comment]
Makes sense as my restart was actually Mac locking up which would have left the port file in place. I wonder if the install.sh script removed the port or one of the commits improves the error handling?

**Comment by markmathis** [comment]
It was fixed in #19. As you said, another solution is to build the current master branch.

**Comment by cflaming** [comment]
I should probably create a pull request with changes to get the install script working better with BSD/Mac to make installing from source easier on Mac. I didn't like the solution I found for fixing `readlink -f`, though, or it would already be done.

**Comment by hyPiRion** [comment]
I'll be happy to merge in a patch making the install script better for BSDs/OS X. :) I also guess it's time to release a new version of Inlein to get rid of these issues, it's a bit silly that I haven't gotten around to do so.

**Comment by RicSwirrl** [comment]
Thanks. Deleting the port file worked for me.  👍

**Comment by basiclaser** [comment]
`rm ~/.inlein/port` did the trick for me

**Comment by hyPiRion** [comment]
@basiclaser: This should be resolved with inlein v0.2.0, so if this is still an issue with the new inlein version I'd love to know.

I'm keeping this open in case someone wants to take a stab at improving the install script for BSD/OS X.

**Comment by hyPiRion** [comment]
I'm closing this, feel free to reopen if it's still an issue.
